### DISCUSSION TITLE
Upgrading to 2.2.4 from 2.1.4

### DISCUSSION THREAD

**Discussion by james-valbuena** [apache/airflow#21835] [General] [answered]
### Apache Airflow version

2.2.4 (latest released)

### What happened

I executed the following command `pip install apache-airflow -U`, to upgrade to the latest version of Airflow. 

This was successful, I can execute `airflow info` and `airflow check` returning 2.2.4 and INFO - Connection successful respectively. 

When I run `airflow webserver`, I receive the following error in the browser:
`
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: session
[SQL: SELECT session.id AS session_id_1, session.session_id AS session_session_id, session.data AS session_data, session.expiry AS session_expiry 
FROM session 
WHERE session.session_id = ?
 LIMIT ? OFFSET ?]
[parameters: ('228b8648-db66-4795-9379-040adb92c08e', 1, 0)]
(Background on this error at: http://sqlalche.me/e/13/e3q8)
`

I have tried 

- airflow db upgrade
- airflow db reset

Neither correct the error, I am really lost to resolve the error and getting nervous as I need my local working for my job.

Any help would be most appreciated. 

### What you expected to happen

Airflow to successfully run after installation 

### Operating System

macOS

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==2.5.0
apache-airflow-providers-ftp==2.0.0
apache-airflow-providers-http==2.0.0
apache-airflow-providers-imap==2.0.0
apache-airflow-providers-mysql==2.1.1
apache-airflow-providers-sqlite==2.0.0

### Deployment

Virtualenv installation

### Anything else

When running `airflow db init`

I see the following error
`
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) duplicate column name: last_scheduling_decision
[SQL: ALTER TABLE dag_run ADD COLUMN last_scheduling_decision TIMESTAMP]
`

When running `airflow db reset`

I receive the following error:

`
sqlite3.OperationalError: table _alembic_tmp_dag already exists
`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Make sure that all your components have the same configuration as explained here: 

https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html

You likely have wrongly configured [SQL_ALCHEMY](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#sql-alchemy-conn) connection (I doubt you use sqlite as a production database).

> Use the same configuration across all the Airflow components. While each component does not require all, some configurations need to be same otherwise they would not work as expected. A good example for that is [secret_key](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-webserver-secret-key) which should be same on the Webserver and Worker to allow Webserver to fetch logs from Worker.

In the future please do not open "airflow issue" if you have a configuration problem. Unless you know the issue is with Airflow - It is strongly recommended to open a discussion instead (You've been warned when you opened it that if you have no reproducible step and clear indication that it's an Airflow Issue.

I convert it into discussion now.

**Comment by james-valbuena** [comment, marked as answer]
@potiuk Many thanks, Jarek for responding on a Saturday, I apologise for not adhering to the correct process. Thank you for amending this as a discussion item. 

I managed to correct the issue by deleting airflow.db and running `airflow db init`

As I am inexperienced in airflow upgrades and felt ill-equipped to resolve. 

Is there a discussion on the best approach to upgrade Airflow? 
I deleted the airflow.db and re-rename `airflow db init`

**Reply by potiuk** [reply]
https://airflow.apache.org/docs/apache-airflow/stable/installation/upgrading.html
